serving/using-external-dns-on-gcp.md

```diff
@@ -0,0 +1,378 @@
+# Using ExternalDNS on Google Cloud Platform to automate DNS setup
+
+[ExternalDNS](https://github.com/kubernetes-incubator/external-dns) is a tool 
+that synchronizes exposed Kubernetes Services and Ingresses with DNS providers.
+
+This doc explains how to set up ExternalDNS within a Knative cluster using
+[Google Cloud DNS](https://cloud.google.com/dns/) to automate the process
+of publishing the Knative domain.
+
+## Set up environtment variables
+
+Run the following command to configure the environment variables
+```shell
+export PROJECT_NAME=<your-google-cloud-project-name>
+
+export CUSTOM_DOMAIN=<your-custom-domain-used-in-knative>
+
+export CLUSTER_NAME=<knative-cluster-name>
+
+export CLUSTER_ZONE=<knative-cluster-zone>
+```
+
+## Set up Kubernetes Engine cluster with CloudDNS read/write permissions
+
+There are two ways to set up a Kubernetes Engine cluster with CloudDNS 
+read/write permissions.
+
+### Cluster with Cloud DNS scope
+
+You can create a GKE cluster with Cloud DNS scope by entering the following 
+command:
+```shell
+gcloud container clusters create $CLUSTER_NAME \
+    --zone=$CLUSTER_ZONE \
+    --cluster-version=latest \
+    --machine-type=n1-standard-4 \
+    --enable-autoscaling --min-nodes=1 --max-nodes=10 \
+    --enable-autorepair \
+    --scopes=service-control,service-management,compute-rw,storage-ro,cloud-platform,logging-write,monitoring-write,pubsub,datastore,"https://www.googleapis.com/auth/ndev.clouddns.readwrite" \
+    --num-nodes=3
+```
+Note that by using this way, any pod within the cluster will have permissions to read/write CloudDNS.
+
+### Cluster with Cloud DNS Admin Service Account credential
+
+1. Create a GKE cluster without Cloud DNS scope by entering the following command:
+```shell
+gcloud container clusters create $CLUSTER_NAME \
+    --zone=$CLUSTER_ZONE \
+    --cluster-version=latest \
+    --machine-type=n1-standard-4 \
+    --enable-autoscaling --min-nodes=1 --max-nodes=10 \
+    --enable-autorepair \
+    --scopes=service-control,service-management,compute-rw,storage-ro,cloud-platform,logging-write,monitoring-write,pubsub,datastore \
+    --num-nodes=3
+```
+
+2. Create a new service account with the project role `dns.admin`:
+```shell
+# Name of the service account you want to create.
```

Comment:
To be more specific, please make each comment a step.

Reply:
Done